tasks/main.yml

```diff
@@ -1,18 +1,18 @@
 ---
-- include: setup-RedHat.yml
-  when: ansible_os_family == 'RedHat'
-
-- include: setup-Debian.yml
-  when: ansible_os_family == 'Debian'
+- name: OS specific setup
+  include: "setup-{{ ansible_os_family }}.yml"
+  when: docker_install
```

Comment:
If you'd like to skip installing docker... then just skip running this role?

I'm not sure what the use case is for running this role but not actually using it to do anything.

Usually, if I have a playbook that needs to do something conditionally (e.g. add a role), I'd add the conditional in the playbook itself, e.g.:

```yaml
- hosts: all
  roles:
    - role: geerlingguy.docker
      when: docker_install
```

(You could use any conditional you'd like there.)